src/EoS/TaubMathews/CPU_EoS_TaubMathews.cpp

```diff
@@ -0,0 +1,325 @@
+//                3. Add "#ifndef __CUDACC__" since this routine is only useful on CPU
+//                4. Do not change the order of AuxArray[]
+//                   --> For example, the dual-energy routines assume AuxArray[0]=GAMMA
+//
+// Parameter   :  AuxArray : Array to be filled up
+//
+// Return      :  AuxArray[]
+//-------------------------------------------------------------------------------------------------------
+#ifndef __CUDACC__
+void EoS_SetAuxArray_TaubMathews( double AuxArray_Flt[], int AuxArray_Int[] )
+{
+   AuxArray_Flt[0] = ( OPT__UNIT ) ? MOLECULAR_WEIGHT * Const_amu / Const_kB * (UNIT_E/UNIT_M)
+
+                                   : MOLECULAR_WEIGHT;
+   AuxArray_Flt[1] = 1.0 / AuxArray_Flt[0];
+} // FUNCTION : EoS_SetAuxArray_TaubMathews
+#endif // #ifndef __CUDACC__
+
+
+
+// =============================================
+// II. Implement EoS conversion functions
+//     (1) EoS_HTilde2Temp_*
+//     (2) EoS_Temp2HTilde_*
+//     (3) EoS_Temper2CSqr_*
+// =============================================
+
+
+
+//-------------------------------------------------------------------------------------------------------
+// Function    :  EoS_GuessHTilde_TaubMathews
+// Description :  Guess a reduced enthalpy for Newton-Raphson iteration
+// Note        :
+// Parameter   :  Con         : Conservative variables
+//                Constant    : The constant on the left side of Eq. A3 in "Tseng et al. 2021, MNRAS, 504, 3298"
+//                AuxArray_*  : Auxiliary arrays (see the Note above)
+//                Table       : EoS tables
+// Return      :  GuessHTilde : guessed reduced enthalpy
+//-------------------------------------------------------------------------------------------------------
+GPU_DEVICE_NOINLINE
+static real EoS_GuessHTilde_TaubMathews( const real Con[], real* const Constant, const double AuxArray_Flt[],
+                                         const int AuxArray_Int[], const real *const Table[EOS_NTABLE_MAX] )
+{
+  real GuessHTilde, Discrimination;
+
+  real Msqr = SQR(Con[1]) + SQR(Con[2]) + SQR(Con[3]);
+  real Dsqr = SQR(Con[0]);
+  real abc = (real)1.0 / Dsqr;
+  real E_D = Con[4] / Con[0];
+  real M_Dsqr = abc * Msqr;
+  real M_D = SQRT( M_Dsqr );
+
+
+  real X = SQRT( E_D*E_D + (real)2.0*E_D );
+  real Y = X + M_D;
+  real Z = X - M_D;
+  *Constant = Y * Z;
+
+
+  real A = (real)437.0 * M_Dsqr + (real)117.0;
+  real B = (real)1.0 + M_Dsqr;
+  real C = (real)43.0*M_Dsqr + (real)63.0;
+
+
+  // Eq. A7 in "Tseng et al. 2021, MNRAS, 504, 3298"
+  Discrimination  = (real)3240000.0 * SQR( B );
+  Discrimination /= SQR( A );
+
+
+  if ( *Constant >= Discrimination )
+  {
+     // Eq. A6 in "Tseng et al. 2021, MNRAS, 504, 3298"
+     GuessHTilde = (real)1.3333333 * SQRT( *Constant );
+  }
+  else
+  {
+     // Eq. A4 in "Tseng et al. 2021, MNRAS, 504, 3298"
+     GuessHTilde  = (real)11.18 * SQRT( B*C* (*Constant) + (real)45.0*B*B );
```

Comment:
Should we change the number `11.18` to `SQRT((real)125.) since the number of digits is insufficient for double precision?

Reply:
GuessHTilde is just an initial guess value before starting root finding. Using 11.18, rather than sqrt(125.0), can slightly improve computing speed.

Reply:
Thanks for the explanation!